components/viz/service/display/record_replay_render.cc

```diff
@@ -260,7 +262,6 @@ void OnPaintFinished(const SkPixmap& pixmap) {
 
 void OnRepaintFinished() {
   CHECK(HasDivergedFromRecording());
-  gRepaintEvent.load()->Signal();
```

Comment:
I wonder if the `gRepaintEvent->Signal` call here can be sent from a previously finished (re-)paint (seeing how its a single global)?

Reply:
I don't think this can be from a "previously finished repaint" - as the call to `OnRepaintFinished` happens from `ProxyImpl::RecordReplayRepaint`, which is triggered from a task-submission from `ProxyMain::RecordReplayRepaint`, which is invoked by the `repaintGraphics` command handler specifically.

There wouldn't have been a `repaintGraphics` issued to the browser before this, so there's no other "prior" code that should be active that would end up in this codepath to trigger this.

Reply:
Its called from `PaintWhenDiverged`. I'm not sure if its possible for a paint to have started in a dirty state but then finishing right after having diverged?

Reply:
You're right it's called from `PaintWhenDiverged`, but `PaintWhenDiverged` is only called under a `repaintGraphics` codepath, and only that leads to the `ProxyMain::RecordReplayPaint => ProxyImpl::RecordReplayPaint` codepath.